x-pack/plugins/transform/public/app/sections/edit_transform/state_management/edit_transform_flyout_state.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  type FC,
+  type PropsWithChildren,
+} from 'react';
+import { configureStore, createSlice } from '@reduxjs/toolkit';
+import { useDispatch, Provider } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { TransformConfigUnion } from '../../../../../common/types/transform';
+
+import { initialize, setApiError, setFormField, setFormSection } from './actions';
+import { type FormFieldsState } from './form_field';
+import { type FormSectionsState } from './form_section';
+import { getDefaultState } from './get_default_state';
+
+// The edit transform flyout uses a redux-toolkit to manage its form state with
+// support for applying its state to a nested configuration object suitable for passing on
+// directly to the API call. For now this is only used for the transform edit form.
+// Once we apply the functionality to other places, e.g. the transform creation wizard,
+// the generic framework code in this file should be moved to a dedicated location.
+
+export interface ProviderProps {
+  config: TransformConfigUnion;
+  dataViewId?: string;
+}
+
+export interface State {
+  apiErrorMessage?: string;
+  formFields: FormFieldsState;
+  formSections: FormSectionsState;
+}
+
+const editTransformFlyoutSlice = createSlice({
+  name: 'editTransformFlyout',
+  initialState: getDefaultState(),
+  reducers: {
+    initialize,
+    setApiError,
+    setFormField,
+    setFormSection,
+  },
+});
+
+const getReduxStore = () =>
+  configureStore({
+    reducer: editTransformFlyoutSlice.reducer,
+  });
+
+const EditTransformFlyoutContext = createContext<ProviderProps | null>(null);
+
+export const EditTransformFlyoutProvider: FC<PropsWithChildren<ProviderProps>> = ({
+  children,
+  ...props
+}) => {
+  const store = useMemo(getReduxStore, []);
+
+  // Apply original transform config to redux form state.
+  useEffect(() => {
```

Comment:
nit: `useMount`, so it's clear the callback should invoke only on mount and  we can omit `// eslint-disable-next-line react-hooks/exhaustive-deps`

Reply:
👍 updated in [31d5d15](https://github.com/elastic/kibana/pull/173861/commits/31d5d15577559420ba7ed24370b705e1cce52227).